### DISCUSSION TITLE
Problem with "url_aac-i2s" example

### DISCUSSION THREAD

**Discussion by romaneromanovich** [pschatzmann/arduino-audio-tools#682] [Q&A] [answered]
Hi Phil.
Thank you very much for your amazing work.

I am trying to build a web radio using "url_aac-i2s" example.
I can't make this example working and constantly getting "Guru Meditation Error...".
In my opinion backtraces point to issues with heap.

I am using ESP32-S3-WROOM-1 with 8MB PSRAM.
All that I changed in example - SSID/PASSWORD and I2S pins.

I've tried to allocate some memory with "heap_caps_malloc_extmem_enable". Situation is better, sound appears, but after few seconds ESP reboots with "Guru Meditation Error...".
Few seconds before reboot I can see in terminal that StreamCopier makes few hops and ESP.getFreeHeap() displays quite big free heap - 190-240kB.

Could you, please, check this example and give me some advice to make it working.
Thank you.

**Comment by pschatzmann** [comment, marked as answer]
Whats the url that you have problems with ?
Did you try to increase the pcm buffer size and the max frame size ?
https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_a_a_c_decoder_helix.html
Further info can be found here: https://github.com/pschatzmann/arduino-libhelix

**Reply by romaneromanovich** [reply]
URL from the example:
url.begin("http://peacefulpiano.stream.publicradio.org/peacefulpiano.aac","audio/aac");

I've just tried to double pcm buffer and max frame sizes and looks like problem was resolved. Now I don't see any problems in terminal and everything works.
I will work with buffer/frame sizes to try to find some optimal values. Thank you for your help.
